amd_openvx_extensions/CMakeLists.txt

```diff
@@ -93,8 +93,8 @@ if(GPU_SUPPORT)
 endif(GPU_SUPPORT)
 
 if (NEURAL_NET)
-    if(GPU_SUPPORT AND miopengemm_FOUND AND miopen_FOUND)
-        if ("${BACKEND}" STREQUAL "OPENCL" AND OpenCL_FOUND)
+    if(GPU_SUPPORT AND miopen_FOUND)
+        if ("${BACKEND}" STREQUAL "OPENCL" AND OpenCL_FOUND AND miopengemm_FOUND)
```

Comment:
@asalmanp can we keep the check for MIOpenGEMM, if not found, not required to add the vx_nn lib?

Reply:
@kiritigowda the previous outer check for MIOpenGEMM was used for both OCL and HIP backends.
in the new code, the miopengemm is only checked if the backend is OCL. 

For OCL backend, the vx_nn lib won't be added if GPU_SUPPORT/MIOpen with OCL backend/MIOpenGEMM is not found.
FOR HIP backend, the vx_nn lib won't be added if GPU_SUPPORT/MIOpen with the HIP backend is not found.